src/main/kotlin/no/nav/familie/ef/sak/beregning/skolepenger/BeregningSkolepengerService.kt

```diff
@@ -1,33 +1,166 @@
 package no.nav.familie.ef.sak.beregning.skolepenger
 
+import no.nav.familie.ef.sak.behandling.BehandlingService
+import no.nav.familie.ef.sak.beregning.barnetilsyn.roundUp
 import no.nav.familie.ef.sak.infrastruktur.exception.brukerfeilHvis
-import no.nav.familie.ef.sak.vedtak.dto.InnvilgelseSkolepenger
+import no.nav.familie.ef.sak.infrastruktur.exception.feilHvis
+import no.nav.familie.ef.sak.tilkjentytelse.TilkjentYtelseService
+import no.nav.familie.ef.sak.tilkjentytelse.domain.TilkjentYtelse
+import no.nav.familie.ef.sak.vedtak.dto.SkolepengerUtgiftDto
 import no.nav.familie.ef.sak.vedtak.dto.UtgiftsperiodeSkolepengerDto
 import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.Year
+import java.time.YearMonth
+import java.time.temporal.ChronoUnit.MONTHS
+import java.util.UUID
+
+private val ONE_HUNDRED: BigDecimal = 100.toBigDecimal()
+
+/**
+ * Er maksbeløp per skoleår? Eller kan det endre seg midt i?
+ *
+ */
+
+private val maksbeløpPerSkoleår = 68_000
 
 @Service
-class BeregningSkolepengerService {
+class BeregningSkolepengerService(
+    private val behandlingService: BehandlingService,
+    private val tilkjentYtelseService: TilkjentYtelseService
+) {
 
-    fun beregnYtelse(innvilgelse: InnvilgelseSkolepenger): List<BeløpsperiodeSkolepengerDto> {
-        return beregnYtelse(innvilgelse.perioder)
+    fun beregnYtelse(
+        utgiftsperioder: List<UtgiftsperiodeSkolepengerDto>,
```

Comment:
Ser greit ut, men for revurdering;
Er det meningen å sende med "tidligere utbetalinger" i utgiftsperiodene? Eller bare nye?